contentcuration/contentcuration/viewsets/contentnode.py

```diff
@@ -556,6 +557,23 @@ def requisites(self, request, pk=None):
             ),
         )
 
+    @detail_route(methods=["get"])
+    def size(self, request, pk=None):
+        if not pk:
+            raise Http404
+        node = self.get_object()
+        files = (
+            File.objects.filter(
+                contentnode__in=node.get_descendants(include_self=True),
+                contentnode__complete=True,
```

Comment:
I have added this change to your code, it's a bit more accurate, in a big channel the difference is almost no appreciable but in small channels it can be significant
```
[content-curation] # SELECT SUM("__col1") FROM (SELECT DISTINCT "contentcuration_file"."checksum" AS Col1, "contentcuration_file"."file_size" AS "__col1" FROM "contentcuration_file" INNER JOIN "contentcuration_contentnode" ON ("contentcuration_file"."contentnode_id" = "contentcuration_contentnode"."id") WHERE ("contentcuration_file"."contentnode_id" IN (SELECT U0."id" AS Col1 FROM "contentcuration_contentnode" U0 WHERE (U0."tree_id" = 25309 AND U0."lft" >= 1 AND U0."lft" <= 7522)))) subquery;
     sum     
-------------
 18768931767
(1 row)

Time: 320,682 ms
[content-curation] # SELECT SUM("__col1") FROM (SELECT DISTINCT "contentcuration_file"."checksum" AS Col1, "contentcuration_file"."file_size" AS "__col1" FROM "contentcuration_file" INNER JOIN "contentcuration_contentnode" ON ("contentcuration_file"."contentnode_id" = "contentcuration_contentnode"."id") WHERE ("contentcuration_file"."contentnode_id" IN (SELECT U0."id" AS Col1 FROM "contentcuration_contentnode" U0 WHERE (U0."tree_id" = 25309 AND U0."lft" >= 1 AND U0."lft" <= 7522)) AND "contentcuration_contentnode"."complete" = True)) subquery;
     sum     
-------------
 18747313583
```

Reply:
Thanks - I guess I took too long to address your comments!

I think we can merge this, but had better keep the issue open, as we will still get timeouts for large channels it seems?

Reply:
I agree

Reply:
Updated the PR accordingly.